src/components/MdRipple/MdRipple.vue

```diff
@@ -1,22 +1,35 @@
 <template>
   <div
-    class="md-ripple"
-    :class="rippleClasses"
+    :class="['md-ripple', rippleClasses]"
     @touchstart.passive.stop="touchStartCheck"
     @touchmove.passive.stop="touchMoveCheck"
-    @mousedown.passive.stop="startRipple">
+    @touchend.passive.stop="clearWave"
+    @mousedown.passive.stop="startRipple"
+    @mouseup.passive.stop="clearWave">
     <slot />
 
-    <transition name="md-ripple" @after-enter="clearWave" v-if="!isDisabled">
-      <span class="md-ripple-wave" :class="waveClasses" :style="waveStyles" v-if="animating" ref="rippleWave" />
-    </transition>
+    <transition-group name="md-ripple" v-if="!isDisabled">
+      <span v-for="(ripple, index) in ripples" :key="'ripple'+index" :class="['md-ripple-wave', waveClasses]" :style="ripple.waveStyles" />
+    </transition-group>
   </div>
 </template>
 
 <script>
   import raf from 'raf'
   import MdComponent from 'core/MdComponent'
 
+  const debounce = function (fn, delay) {
+    var timeoutID = null
+    return function () {
+      clearTimeout(timeoutID)
+      var args = arguments
+      var that = this
+      timeoutID = setTimeout(function () {
+        fn.apply(that, args)
+      }, delay)
+    }
+  }
+
```

Comment:
I think this method could put it into [`/src/core/utils`](https://github.com/vuematerial/vue-material/tree/6a719b5c2421d2efadb3bc522693e438a5a01066/src/core/utils
) and name it `MdDebounce`.

It could be shorter with es6:

```javascript
  const debounce = (fn, delay) => {
    let timeoutID = null
    return (...args) => {
      clearTimeout(timeoutID)
      timeoutID = setTimeout(() => fn(...args), delay)
    }
  }
```